internal/resource/config/oauth/issuer/oauth_issuer_data_source.go

```diff
@@ -0,0 +1,123 @@
+package oauthissuer
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &oauthIssuerDataSource{}
+	_ datasource.DataSourceWithConfigure = &oauthIssuerDataSource{}
+)
+
+// Create a Administrative Account data source
+func NewOauthIssuerDataSource() datasource.DataSource {
+	return &oauthIssuerDataSource{}
+}
+
+// oauthIssuerDataSource is the datasource implementation.
+type oauthIssuerDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type oauthIssuerDataSourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	CustomId    types.String `tfsdk:"custom_id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+	Host        types.String `tfsdk:"host"`
+	Path        types.String `tfsdk:"path"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *oauthIssuerDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schemaDef := schema.Schema{
+		Description: "Describes an OAuth Issuer.",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"description": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"host": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"path": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+		},
+	}
+	id.ToDataSourceSchema(&schemaDef, false, "The persistent, unique ID for the virtual issuer. It can be any combination of [a-zA-Z0-9._-]. This property is system-assigned if not specified.")
+	id.ToDataSourceSchemaCustomId(&schemaDef, true, true, "The persistent, unique ID for the virtual issuer. It can be any combination of [a-zA-Z0-9._-]. This property is system-assigned if not specified.")
```

Comment:
Could remove the last sentence here since this is required for the data source. Also, don't forget to switch from custom_id given your other review.

Reply:
description updated